### PR TITLE
User id migration

### PR DESCRIPTION
**Pull request by rekire**
Our current migration with the id token does not work.
FYI: Somehow an old commit regarding a ssml-builder migration went into this PR.

### CODE REVIEW

File: core/src/main/kotlin/org/rewedigital/dialog/model/google/UserStorageHolder.kt [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+interface UserStorageHolder {
+    var userStorage: String?
+}
+
+val UserStorageHolder.userData: DataStorage
+get() = DataStorage(this)
+
+class DataStorage(private val userData : UserStorageHolder) : LinkedHashMap<String, String>() {
+    init {
```
Comment (MrDyro) [inline core/src/main/kotlin/org/rewedigital/dialog/model/google/UserStorageHolder.kt]: Why don't you use gson or jackson to parse the userdata object?

File: ssml-plugin/build.gradle [hunk 2]
```diff
@@ -10,7 +10,7 @@ apply from: '../docu.gradle'
 dependencies {
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8"
     implementation project(':core')
-    implementation project(':ssml')
+    implementation project(':ssml-builder')
```
Comment (MrDyro) [inline ssml-plugin/build.gradle:13]: What's the reason of this change?

File: core/src/main/kotlin/org/rewedigital/dialog/model/google/UserStorageHolder.kt (hunk 1 above)
Comment (rekire) [reply core/src/main/kotlin/org/rewedigital/dialog/model/google/UserStorageHolder.kt]: AFIK both are not a dependency right now and that small JSON is very easy to create by hand.

File: core/src/main/kotlin/org/rewedigital/dialog/model/google/UserStorageHolder.kt [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        super.remove(key).also {
+            userData.userStorage = asJson()
+        }
+
+    override fun remove(key: String, value: String)=
+        super.remove(key, value).also {
+            userData.userStorage = asJson()
+    }
+
+    val userId: String
```
Comment (MrDyro) [inline core/src/main/kotlin/org/rewedigital/dialog/model/google/UserStorageHolder.kt]: For migration it would be nice to look if there is still an userId. You could save the current userId from the user object to the userdata and if there is no userId you can create one like you already do.

File: ssml-plugin/build.gradle (hunk 2 above)
Comment (rekire) [reply ssml-plugin/build.gradle:13]: This is an old change which I forgot to push. This is related to the way how the modules are pushed to jcenter.

File: core/src/main/kotlin/org/rewedigital/dialog/model/google/UserStorageHolder.kt (hunk 3 above)
Comment (rekire) [reply core/src/main/kotlin/org/rewedigital/dialog/model/google/UserStorageHolder.kt]: Good point, I will check how I can achieve that
